bson/json_util.py

```diff
@@ -869,9 +845,31 @@ def default(obj: Any, json_options: JSONOptions = DEFAULT_JSON_OPTIONS) -> Any:
         elif json_options.datetime_representation == DatetimeRepresentation.LEGACY:
             return {"$date": str(int(obj))}
         return {"$date": {"$numberLong": str(int(obj))}}
-    if json_options.strict_number_long and isinstance(obj, Int64):
+    elif json_options.strict_number_long and type_marker == 18:  # Int64
         return {"$numberLong": str(obj)}
-    if isinstance(obj, (RE_TYPE, Regex)):
+    elif type_marker == 255:  # MinKey
+        return {"$minKey": 1}
+    elif type_marker == 127:  # MaxKey
+        return {"$maxKey": 1}
+    elif type_marker == 17:  # Timestamp
+        return {"$timestamp": {"t": obj.time, "i": obj.inc}}
+    elif type_marker == 13:  # Code
+        if obj.scope is None:
+            return {"$code": str(obj)}
+        return {"$code": str(obj), "$scope": _json_convert(obj.scope, json_options)}
+    elif type_marker == 5:  # Binary
+        return _encode_binary(obj, obj.subtype, json_options)
+    elif type_marker == 19:  # Decimal128
```

Comment:
Nice work! I suspect this could be made even faster by switching to a lookup table, something like:
```python
_encoders = {
    19: lambda obj: {"$numberDecimal": str(obj)},
    ...
}
...

    return _encoders.get(type_marker, raise_type_error)(obj)
```

I believe this is part of what Bernie was suggesting in the ticket.